src/main/java/com/jagrosh/vortex/commands/moderation/SlowmodeCmd.java

```diff
@@ -0,0 +1,130 @@
+    {
+
+        if(event.getArgs().isEmpty())
+        {
+            int slowmodeDuration = vortex.getDatabase().tempslowmodes.timeUntilDisableSlowmode(event.getTextChannel());
+            int slowmodeTime = event.getTextChannel().getSlowmode();
+
+            if(slowmodeTime <= 0)
+            {
+                event.reply("Slowmode is disabled.");
+                return;
+            }
+
+            if(slowmodeDuration <= 0)
+                event.reply("Slowmode is enabled with 1 message every "+FormatUtil.secondsToTimeCompact(slowmodeTime)+".");
+            else
+                event.reply("Slowmode is enabled with 1 message every "+FormatUtil.secondsToTimeCompact(slowmodeTime) +
+                        " for "+FormatUtil.secondsToTimeCompact(slowmodeDuration)+".");
+            return;
+        }
+
+        String args = event.getArgs();
+
+        if(args.equals("0") || args.toLowerCase().equals("off"))
+        {
+            vortex.getDatabase().tempslowmodes.clearSlowmode(event.getTextChannel());
+            event.getTextChannel().getManager().setSlowmode(0).queue();
+            event.replySuccess("Disabled slowmode!");
+            return;
+        }
+
+        String[] split = args.split("\\|",2);
+
+        int slowmodeTime = OtherUtil.parseTime(split[0]);
+        if(slowmodeTime == -1)
+        {
+            event.replyError("Invalid slowmode time!");
+            return;
+        }
+        if(slowmodeTime > MAX_SLOWMODE)
+        {
+            event.replyError("You can only enable slowmode for up to 6 hours!");
+            return;
+        }
+        if(slowmodeTime < -1)
+        {
+            event.replyError("Slowmode cannot use negative time!");
+            return;
+        }
+
+        int slowmodeDuration = split.length == 1 ? 0 : OtherUtil.parseTime(split[1]);
+        if(slowmodeDuration == -1)
+        {
+            event.replyError("Invalid slowmode duration time!");
+            return;
+        }
+        if(slowmodeDuration < -1)
+        {
+            event.replyError("Slowmode duration cannot use negative time!");
+            return;
+        }
+
+        event.getTextChannel().getManager()
+                .setSlowmode(slowmodeTime)
+                .reason("Enabled by "+event.getAuthor().getAsTag()+" ("+event.getAuthor().getId()+")")
+                .queue();
+
+        if(slowmodeDuration > 0)
+        {
+            vortex.getEventWaiter().waitForEvent(
+                    TextChannelUpdateSlowmodeEvent.class,
+                    c -> c.getChannel().getId().equals(event.getTextChannel().getId()),
+                    ev -> vortex.getDatabase().tempslowmodes.setSlowmode(event.getTextChannel(), Instant.now().plus(slowmodeDuration, ChronoUnit.SECONDS)),
+                    10, TimeUnit.SECONDS,
+                    () -> vortex.getDatabase().tempslowmodes.setSlowmode(event.getTextChannel(), Instant.now().plus(slowmodeDuration, ChronoUnit.SECONDS)));
```

Comment:
I'm not sure if you like using event waiters for this case. If not, I may have something else in my mind

Reply:
I think your solution here is interesting, but for the sake of being readable and simple (since you have this event waiter code here, plus the manual call to it in the listener), it might just make more sense to put the slowmode in the database 10 seconds after the queue success happens (if discord takes longer than 10 seconds to deliver the event after we get a reply from the request, there are bigger problems than remembering the slowmode).
```java
...queue(s -> threadpool.submit(() -> database set, 10, TimeUnit.SECONDS))
```